In [1]:
## Import packages 

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statistics
import seaborn as sns
%matplotlib inline

## Load File

df = pd.read_csv('https://raw.githubusercontent.com/borja876/Thinkful-DataScience-Borja/master/WELLCOME_APCspend2013_forThinkful.csv', encoding='latin1')

In [2]:
## View DataFrame
#df

## Determine the five most common journals and the total articles for each

In [3]:
## View counts of each Journal Entry (as typed)
print(df['Journal title'].value_counts())

PLoS One                                                    92
PLoS ONE                                                    62
Journal of Biological Chemistry                             48
Nucleic Acids Research                                      21
Proceedings of the National Academy of Sciences             19
Human Molecular Genetics                                    18
PLoS Neglected Tropical Diseases                            18
Nature Communications                                       17
PLoS Pathogens                                              15
PLoS Genetics                                               15
Neuroimage                                                  15
NeuroImage                                                  14
PLOS ONE                                                    14
Brain                                                       14
BMC Public Health                                           14
Movement Disorders                                     

In [4]:
#df.sort_values('Journal title')

In [5]:
## Converting whole DataFrame to uppercase in the process
df= df.applymap(lambda x: x.upper() if type(x) == str else x )

In [6]:
## Standardize Entries

df['Journal title'] = df['Journal title'].replace(['PLOSONE', 'PLOS ONE', 'PLOS 1', 'PLOS'], 'PLOS ONE') 
df['Journal title'] = df['Journal title'].replace(['ACTA D', 'ACTA CRYSTALLOGRAPHICA SECTION D', 'ACTA CRYSTALLOGRAPHY D', 'ACTA CRYSTALLOGRAPHICA, SECTION D', 'ACTA CRYSTALLOGRAPHICA SECTION D, BIOLOGICAL CRYSTALLOGRAPHY', 'ACTA CRYSTALLOGRAPHICA SECTION D,  BIOLOGICAL CRYSTALLOGRAPHY'], 'ACTA CRYSTALLOGRAPHICA SECTION D: BIOLOGICAL CRYSTALLOGRAPHY') 
df['Journal title'] = df['Journal title'].replace(['AMERICAN JNL EPIDEMIOLOGY'], 'AMERICAN JOURNAL OF EPIDEMIOLOGY') 
df['Journal title'] = df['Journal title'].replace(['AMERICAN JOURNAL OF MEDICAL GENETICS PART A'], 'AMERICAN JOURNAL OF MEDICAL GENETICS')
df['Journal title'] = df['Journal title'].replace(['ANTIMICROBIAL AGENTS AND CHEMOTHERAPY', 'ANTIMICROBIAL AGFENTS AND CHEMOTHERAPY'], 'ANTIMICROBIAL AGENTS & CHEMOTHERAPY') 
df['Journal title'] = df['Journal title'].replace(['ANGEWANDE CHEMIE', 'ANGEWANDTE CHEMIE INTERNATIONAL EDITION','ANGEW CHEMS INT ED' ], 'ANGEWANDTE CHEMIE')
df['Journal title'] = df['Journal title'].replace(['BEHAVIOUR RESEARCH AND THERAPY'], 'BEHAVIOR RESEARCH & THERAPY')
df['Journal title'] = df['Journal title'].replace(['VIROLOGY JOURNAL'], 'VIROLOGY')

In [7]:
## View final counts of each Journal Entry (as typed)
print(df['Journal title'].value_counts())

PLOS ONE                                                                                 210
JOURNAL OF BIOLOGICAL CHEMISTRY                                                           53
NEUROIMAGE                                                                                29
PLOS PATHOGENS                                                                            24
PLOS GENETICS                                                                             24
NUCLEIC ACIDS RESEARCH                                                                    23
PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES                                           20
PLOS NEGLECTED TROPICAL DISEASES                                                          20
HUMAN MOLECULAR GENETICS                                                                  19
NATURE COMMUNICATIONS                                                                     19
BMC PUBLIC HEALTH                                                     

**The five most common jourals (and the respective number of articles) are: PLOS One (210), Journal of Biological Chemistry (53), Neuroimage (29), PLOS Pathogens (24), and PLOS Genetics (24).**

## Next, calculate the mean, median, and standard deviation of the open-access cost per article for each journal

In [8]:
### Cleaning Article Cost Entries 

In [9]:
## removed currency symbols from cost column

## removed £
df= df.apply(lambda x: x.str.replace('£', ''))

## removed $
df= df.apply(lambda x: x.str.replace('$', ''))

In [10]:
## Replace 999999.00 value with mean

df= df.apply(lambda x: x.str.replace('999999.00', '2015.04'))

In [11]:
## Creating Separate DataFrames for each Journal Title

plos_one_isolated=df.loc[lambda df: df['Journal title'] == 'PLOS ONE', :]

jbc_isolated=df.loc[lambda df: df['Journal title'] == 'JOURNAL OF BIOLOGICAL CHEMISTRY', :]

neuroimage_isolated=df.loc[lambda df: df['Journal title'] == 'NEUROIMAGE', :]

plos_pathogens_isolated=df.loc[lambda df: df['Journal title'] == 'PLOS PATHOGENS', :]

plos_genetics_isolated=df.loc[lambda df: df['Journal title'] == 'PLOS GENETICS', :]

In [12]:
## Calculating Mean, Median, Standard deviation for PLOS One article Costs

## Describe PLOS One costs
plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count       210.000000
mean       1891.382286
std       13229.208992
min         122.310000
25%         853.320000
50%         901.810000
75%        1044.737500
max      192645.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

In [13]:
## Remove Extreme Costs 

r = plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).quantile(0.99)

plos_one_isolated_reduced = plos_one_isolated[plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric) < r]

In [14]:
## Describe Article Costs

plos_one_isolated_reduced['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count     201.000000
mean      937.437612
std       193.968371
min       122.310000
25%       852.430000
50%       897.610000
75%      1040.700000
max      1785.360000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS One Article Cost Statistics <br>
mean:      937.437612 <br>
median:    897.610000 <br>
std:       193.968371**

In [15]:
jbc_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      53.000000
mean     1434.747925
std       415.984485
min       265.670000
25%      1152.720000
50%      1314.530000
75%      1610.310000
max      2501.070000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**Journal of Biological Chemistry Cost Statistics <br>
mean:     1434.747925 <br>
median:      1314.530000 <br>
std:       415.984485**

In [16]:
neuroimage_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      29.000000
mean     2215.168276
std       266.653947
min      1747.160000
25%      2100.540000
50%      2326.430000
75%      2396.570000
max      2503.340000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**Neuroimage Cost Statistics <br>
mean:     2215.168276 <br>
median:      2326.430000 <br>
std:       266.653947**

In [17]:
plos_pathogens_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      24.000000
mean     1609.714583
std       198.700369
min      1254.020000
25%      1440.000000
50%      1600.520000
75%      1760.310000
max      2015.040000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS Pathogens Cost Statistics <br>
mean:     1609.714583 <br>
median:      1600.520000 <br>
std:       198.700369**

In [18]:
plos_genetics_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      24.000000
mean     1674.105000
std       180.284357
min      1394.050000
25%      1484.862500
50%      1718.390000
75%      1778.225000
max      2015.040000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS Genetics Cost Statistics <br>
mean:     1674.105000 <br>
median:      1718.390000 <br>
std:       180.284357**